docs/DeveloperGuide.md

```diff
@@ -1,34 +1,426 @@
+    2.4. [Notes Component](#24-notes-component)<br>
+1. [Implementation](#3-implementation)<br>
+    3.1. [Scheduling Tasks](#31-scheduling-tasks)<br>
+    3.2. [Listing Study Areas](#32-listing-study-areas)<br>
+    3.3. [Operation of Notes](#33-operation-of-notes)<br>
+1. [Testing](#4-testing)<br>
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+[Appendix A: Product Scope](#appendix-a-product-scope)<br>
+[Appendix B: User Stories](#appendix-b-user-stories)<br>
+[Appendix C: Non-Functional Requirements](#appendix-c-non-functional-requirements)<br>
+[Appendix D: Glossary](#appendix-d-glossary)<br>
+[Appendix E: Instruction for Manual Testing](#appendix-e-instructions-for-manual-testing)<br>
 
+## 1. Introduction
 
-## Product Scope
-### Target user profile
+### 1.1. Purpose
 
-{Describe the target user profile}
+The document is meant to teach you on how OrgaNice! was developed and the design and implementations behind the software.
+With this document, you should have a better understanding of the framework of the application. 
 
-### Value proposition
+### 1.2. Target Reader
 
-{Describe the value proposition: what problem does it solve?}
+The target reader of this developer guide are:
 
-## User Stories
+- Project Managers
+- Future Developers
+- Build Verification Testers
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+### 1.3. Brief Description
 
-## Non-Functional Requirements
+OrgaNice! is a task manager integrated with a Study Area search function. You can use it to manage your tasks, be it events or deadlines. On top of that, 
+it is capable of assisting students in finding Study Areas that meets their desired criteria.
 
-{Give non-functional requirements}
+### 1.4. Main Features
 
-## Glossary
+1. Scheduling Tasks
+    - You can manage tasks based on priority for events and countdown for deadlines.
+1. Study Area. For more information of the design and implementation for this feature, click [here](#31-scheduling-tasks)
+    - You can find a Study Area that meets your criteria. The software however have a limited number of supported 
+    criteria. For more information of the design and implementation for this feature, click [here](#32-listing-study-areas)
+1. Notes
+	- You can enter notes based on school modules. Notes support undo and redo operations.
 
-* *glossary item* - Definition
+## 2. Design   
+ {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}    
 
-## Instructions for Manual Testing
+[comment]: # (@@author GanapathySanathBalaji)
+### 2.1. Architecture  
+  
+![Architecture](images/Architecture.png)
```

Comment:
Would it make it clearer if different classes used unique colours or arrows to make it easier to differentiate the associations?